opm/simulators/flow/ReservoirCouplingMaster.cpp

```diff
@@ -50,6 +50,37 @@ ReservoirCouplingMaster::ReservoirCouplingMaster(
 // Public methods
 // ------------------
 
+void ReservoirCouplingMaster::receiveSimulationStartDateFromSlaves() {
+    this->slave_start_dates_.resize(this->num_slaves_);
+    if (this->comm_.rank() == 0) {
+        // Ensure that std::time_t is of type long since we are sending it over MPI with MPI_LONG
+        static_assert(std::is_same<std::time_t, long>::value, "std::time_t is not of type long");
+        for (unsigned int i = 0; i < this->master_slave_comm_.size(); i++) {
+            std::time_t start_date;
+            int result = MPI_Recv(
+                &start_date,
+                /*count=*/1,
+                /*datatype=*/MPI_LONG,
+                /*source_rank=*/0,
```

Comment:
I have never used intercommunicators. 

Why is the source 0?
The communicator consists of the processes of the initial communicator an the slave communicator. Who exactly is 0 here? 

Note we could use MPI_ANY_SOURCE if we do not know the rank and there is only one message for the tag.

Reply:
@blattms Thanks for reviewing this. (I plan to rebase and update this PR soon.) The source_rank of 0 means that we will be receiving messages from rank 0 in the remote group of the inter communicator. This corresponds to rank 0 in the slave process sending messages to the parent communicator. Yes, we could use message tag MPI_ANY_SOURCE but it might be clearer to use specific tag and document the type of message we will be receiving here.